### DISCUSSION TITLE
umi从3.1升级到3.5，defaultSettings和basiclayout都获取不到最新的npm 命令携带的参数是为啥？

### DISCUSSION THREAD

**Discussion by dengsheng** [umijs/umi#10260] [Q&A]
a.js
`
export const  query = process.env.npm_config_query
`
basiclayout.tsx
`
import { query } from 'a'
`
执行 npm run dev --query=test
这个query3.1可以拿到，3.5变成undefined

**Comment by fz6m** [comment]
需要 define 传进去吧，项目内能使用的并不是全部 `process.env` 。

**Reply by dengsheng** [reply]
大佬 你值得define传进去是指怎么操作，不好意思没get。但为啥umi 3.1的时候可以拿到

**Reply by fz6m** [reply]
https://v3.umijs.org/config#define

```ts
  define: {
     'process.env.npm_config_query': process.env.npm_config_query
  }
```

**Reply by dengsheng** [reply]
>好的 谢谢大佬
